01_oop_and_ood/src/Point.ts

```diff
@@ -0,0 +1,47 @@
+export class Point {
+    x: number;
+    y: number;
+
+    constructor()
+    constructor(x: number, y: number)
+    constructor(...args: any[]) {
+        if (!args.length) { 
+            this.x = 0; 
+            this.y = 0; 
+        } else if (typeof args[0] === 'number' && typeof args[1] === 'number') { 
```

Comment:
You lose a purpose of `constructor(x: number, y: number)` if you have `(typeof args[0] === 'number' && typeof args[1] === 'number')`.
Also, you have two constructors which show that you will get zero or two arguments, but instead, you work with an array. You lose all of the purposes of the typings here because you simply can have any type of parameters and any amount of them.

Let’s change to
```
constructor(x = 0, y = 0) {
  this.x = 0; 
  this.y = 0; 
}
```

in this case, you don’t need to check if there are some values in x and y because you have default values. Also, there is no reason to check type, because you already have defined it in line 6.

Reply:
done